File: leetcode/0380-insert-delete-getrandom/solution.cpp
class RandomizedSet {
public:
    /**
     * RUNTIME: 322 ms (75.23%)
     * MEMORY: 97 MB (80.67%)
    */
    RandomizedSet() {
    }

    bool insert(int val) {
        if (hashset.find(val) == hashset.end()) {
            hashset[val] = 0;
            return true;
        }
        else {
            return false;
        }
    }

    bool remove(int val) {
        if (hashset.find(val) != hashset.end()) {
            hashset.erase(val);
            return true;
        }
        else {
            return false;
        }
    }

    int getRandom() {
        int random = rand() % hashset.size() + 1;
        int result, cnt = 0;
        for (auto i : hashset) {
            if (++cnt == random) {
                result = i.first;
                break;
            }
        }
        return result;
    }
private:
    unordered_map <int, int> hashset;
};

File: leetcode/0380-insert-delete-getrandom/solution.py
import random

class RandomizedSet:
    """
    RUNTIME: 571 ms (81.63%)
    MEMORY: 61.4 MB (32.70%)
    """
    def __init__(self):
        self.hashset = {}

    def insert(self, val: int) -> bool:
        if val not in self.hashset:
            self.hashset[val] = 0
            return True
        return False

    def remove(self, val: int) -> bool:
        if val in self.hashset:
            del self.hashset[val]
            return True
        return False

    def getRandom(self) -> int:
        return random.choice(list(self.hashset.keys()))